toolchain/check/cpp/macros.cpp

```diff
@@ -53,20 +55,33 @@ auto TryEvaluateMacroToConstant(Context& context, SemIR::LocId loc_id,
     parser.SkipUntil(clang::tok::eof);
     CARBON_DIAGNOSTIC(
         InCppMacroEvaluation, Error,
-        "failed to evaluate macro Cpp.{0} to a valid constant expression",
+        "failed to parse macro Cpp.{0} to a valid constant expression",
         std::string);
     context.emitter().Emit(loc_id, InCppMacroEvaluation, (*name_str_opt).str());
     return nullptr;
   }
 
   clang::Expr::EvalResult evaluated_result;
-  if (!result_expr->EvaluateAsInt(evaluated_result, sema.getASTContext())) {
-    context.TODO(loc_id, "non-integer constant expression in macro.");
+  if (!result_expr->EvaluateAsConstantExpr(evaluated_result,
+                                           sema.getASTContext())) {
+    context.TODO(loc_id,
+                 "failed to evaluate macro to a valid constant expression");
     return nullptr;
   }
```

Comment:
Given the switch to `EvaluateAsConstantExpr` combined with `ParseConstantExpression` above, is there a way to write a test that can reach the TODO? If it seems to be unreachable by tests, maybe make this a CARBON_CHECK, i.e.:

```suggestion
  CARBON_CHECK(result_expr->EvaluateAsConstantExpr(evaluated_result,
                                           sema.getASTContext()));
```

Granted someone may figure out a way to reach this, but then we can also use that to write a test to verify handling. :)

Reply:
I wasn't able to find a test case for this, so switched to using `CARBON_CHECK`.